### PR TITLE
Completed Client and Server Prototype

### LINKED ISSUE #56: Refactor client and server to handle created Tabletop, game items, and player characters

### PR DESCRIPTION
**Pull request by DucksAlongTheLake**
The Client and Server Prototype.  This allows for a master game table to exist on the server, and for clients to only pass functions to the server and to receive back updated copies on the game table.

Closes #56

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -1,35 +1,81 @@
...
 from settings import *
+import time
+
+# queue of functions that are incoming from players and applying to the current game table
+#
+master_function_queue = multiprocessing.Queue()
+
+# we initialize a game table and GM serverside
+#
+gm1 = user.Player()
```
Comment (wheeler-dylan) [inline server.py]: This will have to be rewritten as user.User() to match new changes in Joint

File: server.py [hunk 2]
```diff
@@ -61,6 +109,22 @@ def threaded_client(connection):
...
+# running on the server and process the master function queue onto the game table
+#
+def master_controller(x):
+    global master_function_queue
+    global table1
+    
+    while True:
+        if not master_function_queue.empty():
+            next_function = master_function_queue.get()
+            next_function.execute_function()
```
Comment (wheeler-dylan) [inline server.py]: Oh this is nice to see. :)

File: server.py (hunk 1 above)
Comment (DucksAlongTheLake) [reply server.py]: Handled in the latest commit.  Thanks for catching that.

**Comment by DucksAlongTheLake** [comment]
I went ahead and just pushed a commit that handled the Player rename to User.  I'm also very happy with the Function Packager as well!  I imagine it will have to be altered just a bit once we behind to handle associated objects, but that shouldn't be too hard and will come as needed.

**Pull request merged**